### PR TITLE
example: fix FieldAbsorberTest

### PR DESCRIPTION
**Pull request by psychocoderHPC**
Using spaces in definitions for the command line or CMake could result in parsing issues.

before this PR `-DPARAM_FIELDSOLVER=ArbitraryOrderFDTDPML< 4 BOOST_PP_COMMA() CurrentInterpolation >` was given to the command line, in some cases the compiler interpreted `4`, `BOOST_PP_COMMA()`, `CurrentInterpolation` and `>` as files because the define is only parsed until the first space.

This issue was the first time seen together with clang++-11 provided by HIP 3.18.

@steindev Do you remember why we used `BOOST_PP_COMMA()` instead of a normal comma? 
Could it be that we used spaces around the comma which produced the issues?

### CODE REVIEW

**Review by PrometheusPi — DISMISSED** [review]

**Comment by psychocoderHPC** [comment]
All compiler except `nvcc` supporting a normal comma in the define :-(

**Review by sbastrakov — DISMISSED** [review]
If I remember correctly, it indeed caused issues with both commas and spaces.

**Comment by sbastrakov** [comment]
Wait @psychocoderHPC , but one combination in CI fails as I guess it does not like the new way.

**Comment by psychocoderHPC** [comment]
> Wait @psychocoderHPC , but one combination in CI fails as I guess it does not like the new way.

Yes, I am currently testing the fix in my HIP PR and will move the change into this PR. I thought using comma is working with all compilers but `nvcc` do not like it.

**Comment by steindev** [comment]
> @steindev Do you remember why we used `BOOST_PP_COMMA()` instead of a normal comma?
> Could it be that we used spaces around the comma which produced the issues?

Yes, a normal comma did not work originally. I think the pre-compiler treated everything after the comma as an additional argument.

File: share/picongpu/examples/FieldAbsorberTest/cmakeFlags [hunk 1]
```diff
@@ -30,68 +30,68 @@
...
+#flags[25]="-DPARAM_OVERWRITES:LIST='-DPARAM_FIELDSOLVER=YeePML<CurrentInterpolation>;-DPARAM_PML_SIZE=16;-DPARAM_PML_KAPPA_MAX=1.0;-DPARAM_PML_ALPHA_MAX=0.2;-DPARAM_DIMENSION=DIM2'"
 
 # Convolutional PML in 2D, default strength, stretching
-#flags[26]="-DPARAM_OVERWRITES:LIST='-DPARAM_FIELDSOLVER=YeePML< CurrentInterpolation >;-DPARAM_PML_SIZE=10;-DPARAM_PML_KAPPA_MAX=10.0;-DPARAM_PML_ALPHA_MAX=0.2;-DPARAM_DIMENSION=DIM2'"
-#flags[27]="-DPARAM_OVERWRITES:LIST='-DPARAM_FIELDSOLVER=YeePML< CurrentInterpolation >;-DPARAM_PML_SIZE=16;-DPARAM_PML_KAPPA_MAX=10.0;-DPARAM_PML_ALPHA_MAX=0.2;-DPARAM_DIMENSION=DIM2'"
+#flags[26]="-DPARAM_OVERWRITES:LIST='-DPARAM_FIELDSOLVER=YeePML<CurrentInterpolation>;-DPARAM_PML_SIZE=10;-DPARAM_PML_KAPPA_MAX=10.0;-DPARAM_PML_ALPHA_MAX=0.2;-DPARAM_DIMENSION=DIM2'"
+#flags[27]="-DPARAM_OVERWRITES:LIST='-DPARAM_FIELDSOLVER=YeePML<CurrentInterpolation>;-DPARAM_PML_SIZE=16;-DPARAM_PML_KAPPA_MAX=10.0;-DPARAM_PML_ALPHA_MAX=0.2;-DPARAM_DIMENSION=DIM2'"
 
 # Unixaxial PML in 3D, default strength, no stretching
```
Comment (steindev) [inline share/picongpu/examples/FieldAbsorberTest/cmakeFlags:80]: Would you mind correcting `Unixaxial` to `Uniaxial` right with this PR? (4 times)

**Review by steindev — COMMENTED** [review]
I Like it. Just one little question :sweat_smile:.

File: share/picongpu/examples/FieldAbsorberTest/cmakeFlags (hunk 1 above)
Comment (psychocoderHPC) [reply share/picongpu/examples/FieldAbsorberTest/cmakeFlags:80]: I will fix it in separate PR, this PR passed now the CI.

**Comment by psychocoderHPC** [comment]
This PR can be merged. The solution is not as nice as the version before but it is a test that does not require be very flexible.

btw: later we should move the test into the test folder.

File: share/picongpu/examples/FieldAbsorberTest/cmakeFlags (hunk 1 above)
Comment (psychocoderHPC) [reply share/picongpu/examples/FieldAbsorberTest/cmakeFlags:80]: PR is open: #3528

**Comment by steindev** [comment]
Ahh I am so stupid. SHould have done it the other way round :facepalm:

**Comment by steindev** [comment]
Should I fix the branch? (Since I merged in the wrong order)

**Comment by psychocoderHPC** [comment]
> Should I fix the branch? (Since I merged in the wrong order) Sorry. I know you want to get this done for the HIP addition

I will solve the conflict.

**Review by steindev — APPROVED** [review]

**Pull request merged**